### ISSUE TITLE
Ids gets mixed up with the previous table's ID

### ISSUE THREAD

**Issue by NicksonYap** [staudenmeir/eloquent-has-many-deep#9]
Hi, 

I have this example in the `Address` model

`Address` has `BillingAccount` through `Account`
Address -> Account -> BillingAccount

However, the ID return on the Address array is of the Account array instead.

```php

 public function address_with_billing_account()
    {
        return $this->hasManyDeep(
            BillingAccount::class,
            [
                Account::class,
            ],
            [
                'id', //id of account in account table
                'id',  //id of billing account in billing account table
            ],
            [
                'account_id', //id of account in address table
                'billing_account_id', //id of billing account in account table
            ]
            );
            
        // equivalent:
        // return $this->account->billing_account();
    }

```

It seems to be related to the fact that joins is being used, but then a select query used is of the Account id, that's why Address id gets overridden (because same column name)

Has anyone faced this issue?

Thanks

**Comment by staudenmeir** [comment]
Please post the migrations of all three tables.

**Comment by staudenmeir** [comment]
This is an issue of the underlying `HasManyThrough` relationship and will be fixed in Laravel 5.8: https://github.com/laravel/framework/pull/25812

Until then, you can use this workaround (only for lazy loading):

```php
$address->address_with_billing_account()->getQuery()->get(['billing_accounts.*'])
```

**Comment by NicksonYap** [comment]
@staudenmeir 

To clarify the issue:

---
Eager loaded from `hasManyDeep()`:

![image](https://user-images.githubusercontent.com/5380731/51154772-27a5d180-18b0-11e9-8edc-d8e4b1cd7d8c.png)

---

However, eager loaded ID for the BillingAccount is wrong:

BillingAccount Id should be 1, not 2.
2 is the Account's Id.

![image](https://user-images.githubusercontent.com/5380731/51154800-3ee4bf00-18b0-11e9-9d6c-7a0ee0fb762b.png)


---

I discovered a pattern:

Those that has uses only primary keys `id` on top will cause this exact eager loading issue

It's possible there are `id` conflicts

**Comment by NicksonYap** [comment]
@staudenmeir 

Oops, just saw ur reply after posting.

Where is the suggested code supposed to be placed?
In the controller?


> This is an issue of the underlying `HasManyThrough` relationship and will be fixed in Laravel 5.8: [laravel/framework#25812](https://github.com/laravel/framework/pull/25812)
> 
> Until then, you can use this workaround (only for lazy loading):
> 
> ```
> $address->address_with_billing_account()->getQuery()->get(['billing_accounts.*'])
> ```
